### ISSUE TITLE
 BOOST_HEADER_DEPRECATED("<iterator>")

### ISSUE THREAD

**Issue by advaithm** [rose-compiler/rose#142]
I the latest version of boost the iterator header has been deprecated.  rose needs to be updated to no longer use this header.

**Comment by matzke1** [comment]
From boost-1.74.0, I get this warning (typical of many, many others):

```
  16% 1409) [31.801s] _build/tests/smoke/unit/Utility: tool_CXX testSha256Builtin.C
In file included from testSha256Builtin.C:1:
In file included from ../../../../src/rose.h:147:
In file included from ../../../../src/midend/astProcessing/AstDOTGenerationImpl.C:18:
In file included from ../../../../src/backend/asmUnparser/AsmUnparser_compat.h:5:
In file included from ../../../../src/backend/asmUnparser/AsmUnparser.h:10:
In file included from ../../../../src/midend/BinaryAnalysis/BinaryControlFlow.h:12:
In file included from /home/matzke/.spock/var/installed/wisconsin/b067c16b/boost-nopy/include/boost/graph/adjacency_list.hpp:255:
In file included from /home/matzke/.spock/var/installed/wisconsin/b067c16b/boost-nopy/include/boost/graph/detail/adjacency_list.hpp:37:
In file included from /home/matzke/.spock/var/installed/wisconsin/b067c16b/boost-nopy/include/boost/graph/adjacency_iterator.hpp:13:
/home/matzke/.spock/var/installed/wisconsin/b067c16b/boost-nopy/include/boost/detail/iterator.hpp:13:1: warning: This header is deprecated. Use <iterator> instead. [-W#pragma-messages]
BOOST_HEADER_DEPRECATED("<iterator>")
^
/home/matzke/.spock/var/installed/wisconsin/b067c16b/boost-nopy/include/boost/config/header_deprecated.hpp:23:37: note: expanded from macro 'BOOST_HEADER_DEPRECATED'
# define BOOST_HEADER_DEPRECATED(a) BOOST_PRAGMA_MESSAGE("This header is deprecated. Use " a " instead.")
                                    ^
/home/matzke/.spock/var/installed/wisconsin/b067c16b/boost-nopy/include/boost/config/pragma_message.hpp:24:34: note: expanded from macro 'BOOST_PRAGMA_MESSAGE'
# define BOOST_PRAGMA_MESSAGE(x) _Pragma(BOOST_STRINGIZE(message(x)))
                                 ^
<scratch space>:78:2: note: expanded from here
 message("This header is deprecated. Use " "<iterator>" " instead.")
 ^
1 warning generated.
```

There is nothing we can do about this. The warning is coming from inside boost headers.

If you want to submit an actual warning, please re-open this issue and due so.
